helper/tlsutil/generate.go

```diff
@@ -93,45 +118,74 @@ func GenerateCA(opts CAOpts) (string, string, error) {
 		}
 	}
 
-	id, err := keyID(signer.Public())
+	id, err = keyID(signer.Public())
 	if err != nil {
 		return "", "", err
 	}
 
-	sn := opts.Serial
 	if sn == nil {
 		var err error
 		sn, err = GenerateSerialNumber()
 		if err != nil {
 			return "", "", err
 		}
 	}
-	name := opts.Name
-	if name == "" {
-		name = fmt.Sprintf("Nomad Agent CA %d", sn)
-	}
 
-	days := opts.Days
-	if opts.Days == 0 {
-		days = 365
+	if opts.IsEmpty() && opts.Days == 0 {
+		opts.Name = fmt.Sprintf("Nomad Agent CA %d", sn)
+		opts.Days = 1825
+		opts.Country = "US"
+		opts.PostalCode = "94105"
+		opts.Province = "CA"
+		opts.Locality = "San Francisco"
+		opts.StreetAddress = "101 Second Street"
+		opts.Organization = "HashiCorp Inc."
+		opts.OrganizationalUnit = "Nomad"
+	} else if opts.IsEmpty() && opts.Days != 0 {
+		opts.Name = fmt.Sprintf("Nomad Agent CA %d", sn)
+		opts.Country = "US"
+		opts.PostalCode = "94105"
+		opts.Province = "CA"
+		opts.Locality = "San Francisco"
+		opts.StreetAddress = "101 Second Street"
+		opts.Organization = "HashiCorp Inc."
+		opts.OrganizationalUnit = "Nomad"
```

Comment:
The only difference between the two conditionals is the days; could we therefore add a check for the days inside the `IsEmpty` conditional?